skimage/exposure/_adapthist.py

```diff
@@ -326,36 +308,32 @@ def interpolate(image, slices, maps, lut):
 
     Notes
     -----
-    This function calculates the new graylevel assignments of pixels within
-    a submatrix of the image. This is done by bilinear interpolation between
-    2^image.ndim different adjacent mappings in order to eliminate boundary
-    artifacts.
+    This function calculates the new graylevel assignments of pixels
+    within a submatrix of the image. This is done by multilinear
+    interpolation between 2^image.ndim different adjacent mappings
+    in order to eliminate boundary artifacts.
     """
-    norm = np.product([slices[dim].size
-                       for dim in range(image.ndim)])  # Normalization factor
+    region = tuple([s for s in slices])
+    view = image[region]
 
     # interpolation weight matrices
-    coeffs = np.meshgrid(*tuple([np.arange(slices[dim].size)
-                                 for dim in range(image.ndim)]), indexing='ij')
-    coeffs = [coeff.transpose() for coeff in coeffs]
+    coeffs = np.meshgrid(*tuple([np.arange(s) for s in view.shape[::-1]]),
+                         indexing='ij')
+    coeffs = [np.transpose(c) for c in coeffs]
```

Comment:
I think it is equivalent to :
```suggestion
    coeffs = np.meshgrid(*tuple([np.arange(s) for s in view.shape[::-1]]),
                         indexing='xy')
```

Reply:
Hmm I'm a bit confused about this. I get an error when using the `xy` line. https://docs.scipy.org/doc/numpy/reference/generated/numpy.meshgrid.html says:

> In the 2-D case with inputs of length M and N, the outputs are of shape (N, M) for ‘xy’ indexing and (M, N) for ‘ij’ indexing. In the 3-D case with inputs of length M, N and P, outputs are of shape (N, M, P) for ‘xy’ indexing and (M, N, P) for ‘ij’ indexing.

Seems like the `ij` indexing might be better for the general n-dim case?

Reply:
You are right, I only tested for 2D case, sorry :pray: